app/src/main/java/com/nevilleantony/prototype/downloadmanager/DownloadRepo.java

```diff
@@ -0,0 +1,119 @@
+package com.nevilleantony.prototype.downloadmanager;
+
+import android.content.Context;
+import android.os.Environment;
+import android.util.Log;
+
+import com.nevilleantony.prototype.storage.DownloadsModel;
+import com.nevilleantony.prototype.storage.StorageApi;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class DownloadRepo {
+    private static final String PATH = Environment.getExternalStorageDirectory().getAbsolutePath() + "/Prototype/";
+    private static DownloadRepo downloadRepo = null;
+    private Map<String, FileDownload> downloadMap = new HashMap<>();
+    private StorageApi stgApi = new StorageApi();
+
+    private DownloadRepo(Context context) {
+        File path = new File(PATH);
+        if (!path.exists()) {
+            path.mkdir();
+        }
+        stgApi.setContext(context);
+        List<DownloadsModel> downloadsModelList = stgApi.getAll();
+        for (DownloadsModel model : downloadsModelList) {
+            Log.d("ID", "" + model.getId());
+            FileDownload fileDownload = new FileDownload(
+                    model.getId(),
+                    model.getFile_url(),
+                    model.getRange(),
+                    model.getMin_range(),
+                    model.getMax_range(),
+                    model.getSize()
+            );
+            downloadMap.put(model.getId(), fileDownload);
+        }
+    }
+
+    public static DownloadRepo getInstance(Context context) {
+        if (downloadRepo == null) {
+            downloadRepo = new DownloadRepo(context);
+        }
+        return downloadRepo;
+    }
+
+    public static File createFile(String groupId, String partNo) {
+        File temp = new File(PATH + groupId + File.separator + partNo);
+        File folder = new File(PATH + groupId);
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+        if (!temp.exists()) {
+            try {
+                temp.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+                return null;
+            }
+        }
+        return temp;
+    }
+
+    private static boolean checkFile(String path, String file_name) {
+        File file = new File(path + File.separator + file_name);
+        if (!file.exists()) return false;
+        return true;
+    }
+
+    public Map<String, FileDownload> getDownloadMap() {
+        return downloadMap;
+    }
```

Comment:
No don't provide access to the table from outside, the only thing accessible from outside should be a reference to a single/list of `FileDownload`